tycho/db/event/__init__.py

```diff
@@ -43,9 +48,18 @@ def __init__(self, collection):
 
     async def update_by_id(self, id, update_doc: ModelEvent, insert: bool = False):
         new_data = serialize_to_db_event(update_doc)
-        result = await self.collection.replace_one(
-            {"_id": id}, new_data, upsert=insert)
-        return result
+
+        retries = 1
+        while retries >= 0:
+            try:
+                result = await self.collection.replace_one(
+                    {"_id": id}, new_data, upsert=insert)
+                return result
+            except DuplicateKeyError:
+                await asyncio.sleep(WAIT_TIME_SECONDS)
```

Comment:
should this be a random wait? I guess it's unusual for 3 operations to be identical, but in that situation we have both remaining retries occurring within one second. It would be better to be random between 0 to WAIT_TIME_SECONDS.

Reply:
Updated the wait time to be a random value between 1 to 5 seconds.

Reply:
My Bad. Thanks for clarifying. Updated the wait to be random between 0 and 500ms.